### ISSUE TITLE
Questions about agent's Suspend

### ISSUE THREAD

**Issue by annevk** [tc39/ecma262#1443]
I was looking at https://tc39.github.io/ecma262/#sec-suspend and agents in general and had some questions. (Interchanging signifiers and agents seems to happen elsewhere too.)

1. It's unclear what the calling agent is.
2. "Assert: W is on the list of waiters in WL." does not make sense since WL contains a list of agents, whereas W is merely a signifier for an agent.
3. "suspend W", "W can notify", "If W was notified"; again, W is a signifier, not an agent.
4. "If W was notified explicitly by another agent calling NotifyWaiter(WL, W)" isn't this typically handled via state?

cc @lars-t-hansen

**Comment by syg** [comment]
Good catch on incorrectly using Agent Signifier and Agent interchangeably. Should make an editorial PR when I get some time. I think the intention here is that whenever Agents are manipulated like they were values, Agent Signifiers should be used.

1. The sole caller of Suspend is `Atomics.wait`, which always passes the signifier of "surrounding agent" as W. I agree it's misleading to take a W argument at all, if the action is always to suspend the current agent. Indeed, that W is the surrounding agent's signifier is actually asserted. It should stop being a parameter.
1. Per above, I think this should be changed to a list of Agent Signifiers.
1. How do you feel about language like "the agent whose Agent Record's [[Signifier]] is W"?
1. What state are you referring to here? There's no spec algorithm handling of the suspended state AFAIK in ecma262. The language has informal English. In NotifyWaiter, for instance, it says "wake agent W".

Drive-by: I see that search and replacing wake to notify mangled "W can wake" into "W can notify".

**Comment by annevk** [comment]
> 3\. How do you feel about language like "the agent whose Agent Record's [[Signifier]] is W"?

Sounds acceptable. I'd prefer there to be a way to go from the surrounding agent to the encompassing agent cluster (or whatever you wanna call it) and then use the signifier as a key in a map that agent cluster holds. It seems that's roughly what you want to do here. That would also mean that signifier isn't required to be globally unique. (If you use the language you suggested you might want to update "Uniquely identifies the agent within its agent cluster." to say "Uniquely identifies the agent." or some such.)

> 4\. What state are you referring to here?

Mostly that more explicit language would be preferable, to me. This kind of language makes it very hard to follow what is going on.
